Alpha discard

bonsairobo opened the pull request:
# Objective

I want to call the `pbr_functions::alpha_discard` shader function from my own material shader, but it only takes a `StandardMaterial` parameter.

## Solution

This PR replaces the `StandardMaterial` parameter with less restrictive ones so my shader can call it.

---

## Changelog

- The signature of `pbr_functions::alpha_discard` changed by replacing `StandardMaterial` with only the `flags` and `alpha_cutoff` fields.

## Migration Guide

Replace this:

```wgsl
pbr_functions::alpha_discard(material, ...);
```

with this:

```wgsl
pbr_functions::alpha_discard(material.flags, material.alpha_cutoff, ...);
```

Commit:
extract reusable prepass_alpha_discard

This renames the old prepase_alpha_discard to
prepass_sample_color_and_alpha_discard to indicate
that it accessing PBR-specific texture bindings.

Commit:
refactor alpha_discard params

janhohenheim commented:
Note that this breaks all material extensions, including the example one.

janhohenheim reviewed (changes requested):
Note that this breaks all material extensions, including the example one.
As such, `assets/shaders/extended_material.wgsl` also needs updating.

Commit:
fix usage of alpha_discard in example shaders

Updates extended_material.wgsl and custom_clustered_decal.wgsl to use
the new alpha_discard signature.

andriyDev reviewed (commented):
I tested this out locally and it seems good, the code also looks fine. However, there appear to be merge conflicts. I'd approve after the conflicts are addressed.
